test/integration/apiserver/cel/validatingadmissionpolicy_test.go

```diff
@@ -0,0 +1,918 @@
+										"CREATE",
+									},
+									Rule: admissionregistrationv1.Rule{
+										APIGroups: []string{
+											"",
+										},
+										APIVersions: []string{
+											"*",
+										},
+										Resources: []string{
+											"namespaces",
+										},
+									},
+								},
+							},
+						},
+					},
+					Validations: []admissionregistrationv1alpha1.Validation{
+						{
+							Expression: "object.nonExistentProperty == 'someval'",
+						},
+					},
+					FailurePolicy: &failurePolicy,
+				},
+			},
+			policyBinding: &admissionregistrationv1alpha1.ValidatingAdmissionPolicyBinding{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "validate-namespace-suffix-binding",
+				},
+				Spec: admissionregistrationv1alpha1.ValidatingAdmissionPolicyBindingSpec{
+					PolicyName: "validate-namespace-suffix",
+				},
+			},
+			namespace: &v1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-k8s",
+				},
+			},
+			err: "ValidatingAdmissionPolicy 'validate-namespace-suffix' with binding 'validate-namespace-suffix-binding' denied request: expression 'object.nonExistentProperty == 'someval'' resulted in error: no such key: nonExistentProperty",
+		},
+	}
+
+	for _, testcase := range testcases {
+		t.Run(testcase.name, func(t *testing.T) {
+			defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, genericfeatures.CELValidatingAdmission, true)()
+			server, err := apiservertesting.StartTestServer(t, nil, []string{
+				"--enable-admission-plugins", "ValidatingAdmissionPolicy",
+			}, framework.SharedEtcd())
+			if err != nil {
+				t.Fatal(err)
+			}
+			defer server.TearDownFn()
+
+			config := server.ClientConfig
+
+			client, err := clientset.NewForConfig(config)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			if _, err := client.AdmissionregistrationV1alpha1().ValidatingAdmissionPolicies().Create(context.TODO(), testcase.policy, metav1.CreateOptions{}); err != nil {
+				t.Fatal(err)
+			}
+
+			if _, err := client.AdmissionregistrationV1alpha1().ValidatingAdmissionPolicyBindings().Create(context.TODO(), testcase.policyBinding, metav1.CreateOptions{}); err != nil {
+				t.Fatal(err)
+			}
+
+			// TODO: add retry logic instead
+			time.Sleep(time.Second)
+
+			_, err = client.CoreV1().Namespaces().Create(context.TODO(), testcase.namespace, metav1.CreateOptions{})
+			if err == nil && testcase.err == "" {
+				return
+			}
+
+			if err == nil && testcase.err != "" {
+				t.Logf("actual error: %v", err)
+				t.Logf("expected error: %v", testcase.err)
+				t.Fatal("got nil error but expected an error")
+			}
+
+			if err != nil && testcase.err == "" {
+				t.Logf("actual error: %v", err)
+				t.Logf("expected error: %v", testcase.err)
+				t.Fatal("got error but expected none")
+			}
+
+			if err.Error() != testcase.err {
+				t.Logf("actual validation error: %v", err)
+				t.Logf("expected validation error: %v", testcase.err)
+				t.Error("unexpected validation error")
+			}
```

Comment:
Thoughts on adding the status reason check now? 
```
reason := err.(apierrors.APIStatus).Status().Reason
if reason != testcase.failureReason {
    t.Logf("actual error reason: %v", reason)
    t.Logf("expected failure reason: %v", testcase.failureReason)
    t.Error("unexpected error reason")
}
```

Reply:
Right, I'll mark this as resolved when I add it.